src/snowflake/cli/plugins/stage/diff.py

```diff
@@ -149,12 +151,30 @@ def build_md5_map(list_stage_cursor: SnowflakeCursor) -> Dict[str, str]:
     Returns a mapping of relative stage paths to their md5sums.
     """
     return {
-        strip_stage_name(name): md5
-        for (name, size, md5, modified) in list_stage_cursor.fetchall()
+        strip_stage_name(file["name"]): file["md5"]
+        for file in list_stage_cursor.fetchall()
     }
 
 
-def stage_diff(local_path: Path, stage_fqn: str) -> DiffResult:
+def filter_from_diff(
+    diff: DiffResult, paths_to_sync: Set[str], prune: bool
+) -> List[str]:
+    """Modifies the given diff, keeping only the provided paths. If prune is false, remote-only paths will be empty and the non-removed paths will be returned."""
+    diff.different = [i for i in diff.different if i in paths_to_sync]
+    diff.only_local = [i for i in diff.only_local if i in paths_to_sync]
+    files_not_removed = []
+    if prune:
+        diff.only_on_stage = [i for i in diff.only_on_stage if i in paths_to_sync]
+    else:
+        files_not_removed = [i for i in diff.only_on_stage if i in paths_to_sync]
```

Comment:
Lines 167 and 169 rely on the same list comprehension. Consider assigning it to a placeholder/variable before checking for prune, to reduce duplicating code.